### PR TITLE
feat: add support for json log formats

### PR DESCRIPTION
**Pull request by nabeelsaabna**
Set log format JSON only when: `CONTAINERBASE_LOG_FORMAT` or the falback `LOG_FORMAT` is set to `json` 
Note: case-insensitive 

image

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
please add a test like
https://github.com/containerbase/base/blob/3c7a21fd6acbeff07617f8de6878add21880b2c5/src/cli/utils/logger.spec.ts#L15-L34

File: src/cli/utils/logger.spec.ts [hunk 1]
```diff
@@ -32,4 +32,39 @@ describe('cli/utils/logger', () => {
       },
     );
   });
+
+  test('works - stdout with json', async () => {
+    env.CONTAINERBASE_LOG_FORMAT = 'json';
```
Comment (viceice) [inline src/cli/utils/logger.spec.ts:37]: please delete these new variables like the others in `beforeEach`

**Review by viceice — COMMENTED** [review]
like here

https://github.com/containerbase/base/blob/c23e372f4f0f5e900127e89f82092fc2d54211c9/src/cli/utils/logger.spec.ts#L9-L12

File: src/cli/utils/logger.spec.ts (hunk 1 above)
Comment (nabeelsaabna) [reply src/cli/utils/logger.spec.ts:40]: done

**Pull request merged**